src/util.ts

```diff
@@ -100,80 +101,90 @@ export function showChangelog() {
    └─────────────────┘ */
 
 /**
+ * Convert Markdown to plain text.
  * Remove Markdown syntax (bold, italic, links etc.) in a heading.
- * This function is only used before the `slugify` function of `github` mode.
- * 
+ * This function is only for the `github` and `gitlab` slugify functions.
+ *
  * A Markdown heading may contain Markdown styles, e.g. `_italic_`.
  * It can also have HTML tags, e.g. `<code>`.
- * They should not be passed to the `slugify` function.
+ * They should be converted to their plain text form first.
  *
  * What this function actually does:
- * 1. (Escape syntax like `1.`)
- * 2. `md.render(text)`
- * 3. `getTextInHtml(text)`
- * 4. (Unescape)
+ * 1. Handle a few special cases.
+ * 2. `renderInline(text)`
+ * 3. `getTextInHtml(html)`
  *
- * @param text A Markdown heading
+ * @param text - A Markdown heading content.
  */
-function mdHeadingToPlaintext(text: string) {
+function mdHeadingToPlaintext(text: string): string {
     //// Issue #515
     text = text.replace(/\[([^\]]*)\]\[[^\]]*\]/, (_, g1) => g1);
-    //// Escape leading `1.` and `1)` (#567, #585)
-    text = text.replace(/^([\d]+)(\.)/, (_, g1) => g1 + '%dot%');
-    text = text.replace(/^([\d]+)(\))/, (_, g1) => g1 + '%par%');
```

Comment:
Did you check this? It is kind of special as they are CommonMark syntaxes but...

It occurred to me whether we can properly deal with a heading like `# - pretends to be a list`.

Reply:
They are covered by `renderInline`.

Reply:
Nice!